lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -39,7 +39,19 @@ use virt_utils;
 
 our @EXPORT
   = qw(download_network_cfg prepare_network restore_standalone destroy_standalone restart_libvirtd restart_network restore_guests restore_network
-  destroy_vir_network restore_libvirt_default enable_libvirt_log ssh_setup upload_debug_log check_guest_status save_guest_ip test_network_interface hosts_backup hosts_restore);
+  destroy_vir_network restore_libvirt_default enable_libvirt_log ssh_setup upload_debug_log check_guest_status check_guest_module save_guest_ip test_network_interface hosts_backup hosts_restore);
+
+sub check_guest_module {
+    my ($guest, %args) = @_;
+    my $module = $args{module};
+    if ($guest =~ m/sles-?11/i) {
+        my $status = script_run("ssh root\@$guest \"lsmod | grep $module\"");
+        if ($status != 0) {
+            script_run("ssh root\@$guest modprobe $module", 60);
+            record_soft_failure 'bsc#1167828 - need to load acpiphp kernel module to sles11sp4 guest otherwise network interface hotplugging does not work';
+        }
+    }
+}
```

Comment:
@grisu48 this might be interesting in the kernel smoketest you did.

Reply:
Thanks for pointing this out. I'm not sure if this is something we should do in a general smoketest though.

Reply:
This routine will only work for the `acpiphp` module as intended. If this is run for another module and the check fails it will report `bsc#1167828`. I suggest to either move the soft_failure outside of the routine or don't allow to specify the kernel module as argument.

Reply:
Awesome! Nice work here. 
It is so helpful to me. 
I will updated this as your mentioned

Reply:
`+    if (($guest =~ m/sles-?11/i) && ($module eq "acpiphp")) {`